docs: add servicebus service-to-service correlation userguide to v2.6 feature docs (#467)
arcus-azure/arcus.observability: docs/versioned_docs/version-v2.6.0/02-Guidance/Service-to-service Correlation/use-service-to-service-correlation-in-service-bus.md

---
title: "Use service-to-service correlation in Azure Service Bus solutions"
layout: default
---

# Use service-to-service correlation in Azure Service Bus solutions
The concept of service-to-service correlation is big and complex and spans multiple Arcus libraries. This user-guide will walk through all the available Arcus features that work together to facilitate service-to-service correlation in your Azure Service Bus solutions. For a general introduction, see the [introduction page on service-to-service correlation](index.md).

> To use service-to-service correlation in Azure Service Bus solutions, both the sending and receiving side of the separate components have to be adapted. The correlation will be passed through from one service to another via the Azure Service Bus message application properties which the internal Arcus system will use to link your services together.

Reviewer: ```suggestion
To use service-to-service correlation in Azure Service Bus solutions, both the sending and receiving side of the separate components have to be adapted. The correlation will be passed through from one service to another via the Azure Service Bus message application properties which Arcus will use to link your services together.
```
Isn't that simpler / easier to read ?

Author: Yes, that's true. Thx! 👍 
https://github.com/arcus-azure/arcus.observability/pull/467/commits/7afadc0cf3ab390f9fe62845d5cd1ff0abdc8e02

> The following diagram shows how an Web API calls a Service Bus with correlation tracking:

Reviewer: ```suggestion
The following diagram shows how a Web API calls a Service Bus with correlation tracking:
```

Author: https://github.com/arcus-azure/arcus.observability/pull/467/commits/d3106907ad596c8a06b904ef73e45a95832e8700

4. The `jkl` part of this ID, describes the new parent ID for service B (when service B calls service C, then it will use `jkl` as parent ID)
5. The user receives both the **transaction ID** and **operation parent ID** in their final response.

> 💡 Additional configuration is available to tweak this functionality, see the [dedicated Arcus Messaging feature documentation](https://messaging.arcus-azure.net/Features/message-pumps/service-bus) for more in-depth information on Azure Service Bus correlation.

## Service Bus demonstration
In this user-guide, a fictive API and Service Bus application will be used to represent Service A and Service B of the diagram. Both services will be adapted to use service-to-service correlation.

* **Order API** (Service A): receives an request to order a product, sends the order to process.
* **Order Worker** (Service B): receives the order request and processes the order.

The user interacts with the **Product API** to order their product. Internally, the **Order API** will contact the **Order Worker** to further process the order request.

> > ⚠ Take into account that this sample should only be used for demonstration purposes and does not reflect a fully production-ready implementation. We have chosen to only provide the bare bones of the application so the changes are clear.

Reviewer: ```suggestion
> ⚠ Take into account that this sample should only be used for demonstration purposes and does not reflect a fully production-ready implementation. We have chosen to only provide the bare bones of the application to be able to focus on the changes required for service-to-service correlation.
```

Author: Rather long for such a short purpose, but ok.

Author: https://github.com/arcus-azure/arcus.observability/pull/467/commits/1d7dc7d322057da07d44e7d48fb60b7cb478370e
